search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+ *     results.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ */
+async function saveResultsAndDeletePrevious(searchResultsJson, countryData) {
+  await deleteAncientResults();
+
+  // Parse the JSON string and pass each search result to add to the
+  // countryData object.
+  var currentSearchResults = searchResultsJson.items;
+  try {
+    let avg = 0;
+    if (currentSearchResults == undefined) {
+      return 0;
+    } else {
+      for (var i = 0; i < currentSearchResults.length; i++) {
+        avg = avg + await addSearchResultToCountryData(currentSearchResults[i], countryData);
+        console.log('avg in for loop of saveResults' + avg);
+      }
+      return avg / currentSearchResults.length;
+    }
+  } catch (err) { // Occurs when no search results for that country and topic.
+    console.error('ERROR:', err);
+    countryData = null;
+  }
+}
+
+/** Deletes search results from 7 days ago. */
+async function deleteAncientResults() {
+  const query = datastore.createQuery('CustomSearchTopic').order('timestamp');
+  const [searchResults] = await datastore.runQuery(query);
+
+  // Note: Can't use forEach with await.
+  // Loop through sorted data beginnning with oldest results, delete if older
+  // than a week. Stop when reach results from within a week.
+  for (let i = 0; i < searchResults.length; i++) {
+    if (Date.now() - searchResults[i].timestamp > 7 * 24 * 60 * 60000) {
+      const searchResultKey = searchResults[i][datastore.KEY];
+      await datastore.delete(searchResultKey);
+      console.log(`Custom Search Result ${searchResultKey.id} deleted.`)
+    } else {
+      break;
+    }
+  }
+}
+
+/** 
+ * Creates Datastore item for given topic and country search results. 
+ * @param {string} topic The seach topic the search results are for.
+ * @param {Object} countriesData Object holding all searchResults for all
+ *      countries.
+ */
+async function addTopicToDatastore(topic, countriesData) {
+  const customSearchTopicKey = datastore.key('CustomSearchTopic');
+  // Get the current timestamp in milliseconds.
+  let timestamp = Date.now();
+  const entity = {
+    key: customSearchTopicKey,
+    data: {
+      topic: topic,
+      countries: countriesData,
+      timestamp: Date.now(),
+    },
+  };
+  try {
+    await datastore.save(entity);
+    console.log(`Custom Search Result ${customSearchTopicKey.id} created successfully.`);
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/**
+ * Adds search result object to countryData array.
+ * @param {Object} searchResult Object with information for one search result.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ * @return {Promise} A Promise wrapped around the sentiment score of the result.
+*/
+function addSearchResultToCountryData(searchResult, countryData) {
+  return getSentiment(searchResult)
+    .then(response => response.json())
```

Comment:
I think this needs to be 4 space indented.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then I think two, from these examples?

Reply:
Okay well I actually combined these two into one line, but the .then() on the line after still needs to be decided.
I am not sure what the formatting should be. In the below example I would think it should be 4 spaces.
https://google.github.io/styleguide/jsguide.html#formatting-function-expressions

Reply:
indent 4 spaces when this is a line break; indent 2 when this is breaking for the function body.

1
return getSentiment(searchResult)
xxxx.then(response => response.json())
xxxx.then(result => {
xxxxxxblahblah
xxxx}

2
return getSentiment(searchResult).then(response =>
xxresponse.json()).then((result) => {
xxblahblah
xx}

I personally prefer the first style